File: src/net/fodt/Spec/Text/Span.cs
// FormatFactory.Fodt — Spec.Text.Span — Canonical spec-shaped model class
// spec_qname: text:span
// spec_fact_ref: FACT-FODT-006
namespace FormatFactory.Fodt.Spec.Text;

/// <summary>
/// Spec-shaped model class for the ODF text:span element.
///
/// ODF 1.3 §5.1.5 — text:span is an inline character-style element used to
/// apply character formatting to a run of text within a text:p or text:h element.
/// spec_qname: text:span
/// spec_fact_ref: FACT-FODT-006
/// </summary>
public sealed class Span
{
    /// <summary>The ODF QName for this element. Grounded in ODF 1.3 §5.1.5.</summary>
    public const string SpecQName = "text:span";

    /// <summary>The SAL fact reference for this element (sal-facts-latest.json).</summary>
    public const string SpecFactRef = "FACT-FODT-006";

    /// <summary>The text:style-name attribute value — the character style applied. May be null.</summary>
    public string? StyleName { get; init; }

    /// <summary>The plain-text content of this span element.</summary>
    public string Content { get; init; } = string.Empty;
}

File: src/python/fodt/spec/text/span.py
"""FODT spec Span — canonical implementation of text:span.

spec_qname: text:span
spec_fact_ref: FACT-FODT-006
"""
from __future__ import annotations
from typing import Any, ClassVar


class Span:
    """Canonical implementation of ODF text:span element."""

    spec_qname: ClassVar[str] = "text:span"
    spec_fact_ref: ClassVar[str] = "FACT-FODT-006"

    def __init__(self, data: dict[str, Any]):
        self._data = data

    @property
    def text(self) -> str:
        return self._data.get("text", "")

    @property
    def style_name(self) -> str:
        return self._data.get("style_name", "")

    def to_dict(self) -> dict[str, Any]:
        return dict(self._data)

    def __repr__(self) -> str:
        return f"Span(text={self.text!r})"
